pkg/core/tekton/builder/pipeline_run.go

```diff
@@ -0,0 +1,104 @@
+package builder
+
+import (
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PipelineRunBuilder holds the tekton pipeline definition
+type PipelineRunBuilder struct {
+	PipelineRun v1beta1.PipelineRun
+}
+
+// NewPipelineRunBuilder creates a PipelineRun with default values.
+func NewPipelineRunBuilder(generateName string) *PipelineRunBuilder {
+	b := &PipelineRunBuilder{}
+	b.PipelineRun = v1beta1.PipelineRun{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PipelineRun",
+			APIVersion: "tekton.dev/v1beta1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: generateName,
+		},
+	}
+	return b
+}
+
+// PipelineRunMeta sets the Meta structs of the Pipeline.
+// Any number of MetaOp modifiers can be passed.
+func (b *PipelineRunBuilder) PipelineRunMeta(ops ...MetaOp) {
```

Comment:
Same here: you could strip the prefix `PipelineRun` from the method names.

Reply:
done